prototypes/alleleFrequencies/renderAlleles.py

```diff
@@ -0,0 +1,113 @@
+import vcfAlleleLibrary
+import matplotlib.pyplot as plt
+import numpy as np
+
+"""Given a list of ExAC alleles, a bar chart of allele frequencies is plotted, with 
+subpopulations on the x axis and frequencies on the y axis"""
+
+
+def plotAlleles(alleleList):
+    """Given an alleleList, will produce 2 bar charts rendering allele freq
+    data"""
+    figureNum = 0 #will allow "figure 1" and "figure 2" to be made
+    for allele in alleleList: #allele is an object with attributes
+        figureNum += 1
+        if len(allele.pops) == 5:
+            color = 'cyan'
+            title = '1000 Genomes Frequencies'
+            plotAllele(allele, title, color)
+            truncPlot(allele, title, color)
+        if len(allele.pops) == 7:
+            color = 'salmon'
+            title = 'ExAC Frequencies'
+            plotAllele(allele, title, color)
+            truncPlot(allele, title, color)
+        else:
+            print 'Vcf.gz not recognized'
+
+
+def plotAllele(allele, title, color):
+    """given an allele, will create a bar chart for its allele frequencies"""
+    labels, AFs = makeFrequencyVectors(allele)
+    sigLine = np.full(len(labels), 0.05)
+    xAlign = np.arange(len(labels))
+    fig, ax = plt.subplots()
+    ax.bar(xAlign, AFs, color= color)
+    ax.plot(xAlign, sigLine, color = 'black')
+    ax.set_ylabel('Allele Frequencies')
+    ax.set_title(title)
+    ax.set_xticks(xAlign) #this is the line that made a difference for label alignment
+    ax.set_xticklabels(labels)
+    #ax.set_ylim((0,1))
+    plt.show()
+
+def truncPlot(allele, title, color):
+    labels, AFs = makeFrequencyVectors(allele)
+    sigLine = np.full(len(labels), 0.05)
+    xAlign = np.arange(len(labels))
+    fig, (ax, ax2) = plt.subplots(2, 1, sharex=True)
+    ax.plot(xAlign, sigLine, color = 'black')
+    #ax2.plot(xAlign, sigLine, color = 'black')
+    ax.set_title(title)
+    ax.bar(xAlign, AFs, color= color)
+    ax2.bar(xAlign, AFs, color = color)
+    ax.set_ylim(.02, 1.)
+    ax2.set_ylim(0, .001) #figure out how to let this autofit
+    ax.spines['bottom'].set_visible(False)
+    ax2.spines['top'].set_visible(False)
+    ax.xaxis.tick_top()
+    ax.tick_params(labeltop='off')
+    ax2.xaxis.tick_bottom()
+
+    d = .015
```

Comment:
What does this represent?
Here's a suggestion that will make your life easier: avoid single-letter variable names.  If you merely replace the variable "d" with "dd", you'll thank yourself when and if the day comes when you have to search your file for every instance of the variable...